include/validator/validator_component.h

```diff
@@ -17,10 +71,31 @@ struct ExternDescVisitor {
 };
 
 struct InstanceExprVisitor {
-  Expect<void> operator()(const Instantiate &) {
-    // TODO: Validation of instantiate requires each <importname> in c to
-    // match a name in a with argument (compared as strings) and for the
-    // types to match.
+  InstanceExprVisitor(Context &Ctx) : Ctx(Ctx) {}
+
+  Expect<void> operator()(const Instantiate &Inst) {
+    auto Args = Inst.getArgs();
+    auto ImportMap = getImports(Inst.getComponentIdx());
+
+    for (const auto &[ImportName, ImportDesc] : ImportMap) {
```

Comment:
![image](https://github.com/user-attachments/assets/6d67e327-76db-4d9f-9c54-6d801088ac8b)

This is C++20 syntax, WasmEdge still using 17 for now, you will need to configure your compiler.